CMakeLists.txt

```diff
@@ -395,25 +453,28 @@ if(LIBUV_BUILD_TESTS)
            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 endif()
 
-if(UNIX)
+if (UNIX)
   # Now for some gibbering horrors from beyond the stars...
-  foreach(x ${uv_libraries})
-    set(LIBS "${LIBS} -l${x}")
-  endforeach(x)
+  foreach(lib IN LISTS uv_libraries)
+    list(APPEND LIBS "-l${lib}")
+  endforeach()
+  string(REPLACE ";" " " LIBS "${LIBS}")
+  #Consider setting project version via project() call?
   file(STRINGS configure.ac configure_ac REGEX ^AC_INIT)
-  string(REGEX MATCH [0-9]+[.][0-9]+[.][0-9]+ PACKAGE_VERSION "${configure_ac}")
-  string(REGEX MATCH ^[0-9]+ UV_VERSION_MAJOR "${PACKAGE_VERSION}")
+  string(REGEX MATCH "([0-9]+)[.][0-9]+[.][0-9]+" PACKAGE_VERSION "${configure_ac}")
+  set(UV_VERSION_MAJOR "${CMAKE_MATCH_1}")
```

Comment:
For my edification: why this change?

Reply:
This is here because the CMake regex engine is VERY poorly implemented. It's effectively taken from a public domain posted to a BBS in 1986 and hasn't changed since (it also can't be updated because then behavior would change 😓). Performing two regex calls here is also unnecessary, since after a `REGEX MATCH` call, CMake will set several `CMAKE_MATCH_<n>` variables. By putting a capture group at the front (where the major version is), we can do just the one call and then do a less expensive operation of setting the capture group.

It doesn't affect libuv itself, but anyone who uses `add_subdirectory` to include libuv in their build will get a small speed increase if their build is large enough. 

Though, the fastest thing would be to put the version inside the call to `project()` and update it as necessary.